src/FocusEvent.ts

```diff
@@ -221,6 +283,16 @@ export function setupFocusEvent(win: Window): void {
     return origFocus.apply(this, arguments);
   }
 
+  let activeElement = kwin.document.activeElement as Element | null;
+
+  // If keyborg is created with the focus inside shadow root, we need
+  // to go through the shadows up to make sure all relevant shadows
+  // have focus handlers attached.
+  while (activeElement && activeElement.shadowRoot) {
```

Comment:
do we need to drill down to all shadow roots? if so what about parent shadow roots? do we need to handle them all too?